### PR TITLE
Edits PokemonController to use async/await for promises

### PR DESCRIPTION
**Pull request by Austin-Gray**

### CODE REVIEW

File: server/pokemon_controller.ts [hunk 1]
```diff
@@ -26,17 +26,19 @@ interface Pokemon {
...
-      const meta: { count: string } = { count: res.data.results.length };
-      const items: Pokemon[] = res.data.results;
-      const byId: ById<Pokemon> = {};
-      items.forEach((pokemon, i) => Object.assign(byId, { [i+1]: pokemon }));
-      return { meta, items, byId };
-    })
+  async index() {
+    const res = await axios.get('https://pokeapi.co/api/v2/pokemon?limit=807');
+    const pokedata = res.data;
+    const meta: { count: string } = { count: pokedata.results.length };
```
Comment (themindoverall) [inline server/pokemon_controller.ts]: `count` is not a string is it?

File: server/pokemon_controller.ts [hunk 2]
```diff
@@ -26,17 +26,19 @@ interface Pokemon {
...
+    const items: Pokemon[] = pokedata.results;
+    const byId: ById<Pokemon> = {};
+    items.forEach((pokemon, i) => Object.assign(byId, { [i+1]: pokemon }));
+    return { meta, items, byId };
   }
-  show(request: Hapi.Request, h: Hapi.ResponseToolkit) {
+  async show(request: Hapi.Request, h: Hapi.ResponseToolkit) {
     const search = request.params.poke;
-    return axios.get(`https://pokeapi.co/api/v2/pokemon/${search}`).then(res => res.data )
+    const res = await axios.get(`https://pokeapi.co/api/v2/pokemon/${search}`);
```
Comment (themindoverall) [inline server/pokemon_controller.ts]: What happens if the pokemon isn't found?  While you'll get a 404 response from axios, that won't necessarily translate automatically be a 404 response from your hapi server.

File: server/pokemon_controller.ts (hunk 1 above)
Comment (Austin-Gray) [reply server/pokemon_controller.ts]: `count` is now a number

File: server/pokemon_controller.ts (hunk 2 above)
Comment (Austin-Gray) [reply server/pokemon_controller.ts]: Routes now return a 404 response if they are unable to get pokemon data.

**Pull request merged**